### PR TITLE
layers: Add VK_REMAINING_ARRAY_LAYERS VUs

### PR DESCRIPTION
**Pull request by spencer-lunarg**
For `VK_KHR_maintenance5` added VUs to check when copying images and only one has `VK_REMAINING_ARRAY_LAYERS`

- VUID-VkCopyImageInfo2-srcImage-08794
- VUID-VkCopyImageInfo2-srcImage-08794
- VUID-VkImageBlit-layerCount-08801
- VUID-VkImageBlit2-layerCount-08801
- VUID-VkImageResolve-layerCount-08804
- VUID-VkImageResolve2-layerCount-08804

### CODE REVIEW

File: layers/core_checks/cc_copy_blit_resolve.cpp [hunk 1]
```diff
@@ -1882,20 +1886,37 @@ bool CoreChecks::ValidateCmdCopyImage(VkCommandBuffer commandBuffer, VkImage src
                 // Depth comes from layerCount for 1D,2D resources, from extent.depth for 3D
                 uint32_t src_slices = (src_is_3d ? src_copy_extent.depth : region.srcSubresource.layerCount);
                 uint32_t dst_slices = (dst_is_3d ? dst_copy_extent.depth : region.dstSubresource.layerCount);
-                if (src_slices != dst_slices && src_slices != VK_REMAINING_ARRAY_LAYERS &&
-                    dst_slices != VK_REMAINING_ARRAY_LAYERS) {
-                    const LogObjectList objlist(commandBuffer, srcImage, dstImage);
+                if (src_slices == VK_REMAINING_ARRAY_LAYERS || dst_slices == VK_REMAINING_ARRAY_LAYERS) {
+                    if (src_slices != VK_REMAINING_ARRAY_LAYERS) {
+                        if (src_slices != (dst_image_state->createInfo.arrayLayers - region.dstSubresource.baseArrayLayer)) {
```
Comment (artem-lunarg) [inline layers/core_checks/cc_copy_blit_resolve.cpp:1891]: The VU itself is a bit confusing for me.
> be either VK_REMAINING_ARRAY_LAYERS or equal to the arrayLayers member of the VkImageCreateInfo used to create the **image** minus baseArrayLayer

It's not clear which image is referenced. The code does comparison against opposite image, but other parts of the spec use VK_REMAINING to determine count for current image.

Comment (spencer-lunarg) [reply layers/core_checks/cc_copy_blit_resolve.cpp:1891]: `VK_REMAINING_ARRAY_LAYERS` is really just a way to have the driver use what ever `VkImageCreateInfo::arrayLayers` - `VkImageSubresourceRange::baseArrayLayer` should be

When doing a copy, if my `srcImage` has 3 layers, my `dstImage` needs 3 layers as well (`VUID-vkCmdCopyImage-srcImage-08793`)

if the `srcImage` has `layerCount` of `3` then we need to make sure `VK_REMAINING_ARRAY_LAYERS` for the `dstImage` is implicitly `3` as well

for this, we make sure `dstImage::arrayLayers` - `dstSubresource::baseArrayLayer` is 3

----

now talking out loud, I realize my logic is flipped, will make fix now

Comment (spencer-lunarg) [reply layers/core_checks/cc_copy_blit_resolve.cpp:1891]: I correct myself, in this line of code, we know `dstSubresource::layerCount` is `VK_REMAINING_ARRAY_LAYERS` and now much make sure that the remaining layers of the `dstImage` match the number of "Source layers" (`src_slices`)

Comment (artem-lunarg) [reply layers/core_checks/cc_copy_blit_resolve.cpp:1891]: Yes, it makes sense. Was confused by the spec language.

**Review by artem-lunarg — APPROVED** [review]

**Pull request merged**